src/App.tsx

```diff
@@ -1,26 +1,209 @@
+    postService
+      .getTodos()
+      .then(fetchedTodos => {
+        setTodos(fetchedTodos);
+      })
+      .catch(() => {
+        handleError('Unable to load todos', setErrorMessage);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+  useEffect(() => {
+    if (shouldFocusInput) {
+      setShouldFocusInput(false);
+    }
+  }, [shouldFocusInput]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const itemsLeft = todos.filter(({ completed }) => !completed).length;
+  const haveCompletedTodos = todos.some(({ completed }) => completed);
+  const allCompleted = todos.every(({ completed }) => completed);
+
+  const filteredTodos = todos.filter(task => {
+    switch (status) {
+      case Status.Active:
+        return !task.completed;
+      case Status.Completed:
+        return task.completed;
+      case Status.All:
+      default:
+        return true;
+    }
+  });
+
+  const addTodo = (title: string, setTitle: (title: string) => void) => {
+    const trimmedTitle = title.trim();
+
+    if (!trimmedTitle) {
+      handleError('Title should not be empty', setErrorMessage);
+
+      return;
+    }
+
+    setInputDisabled(true);
+
+    const newTodo: Todo = {
+      id: Math.max(0, Math.max(...todos.map(todo => todo.id))) + 1,
+      title: trimmedTitle,
+      completed: false,
+      userId: USER_ID,
+      status: status,
+    };
+
+    const fakeTodo: Todo = {
+      id: 0,
+      title: trimmedTitle,
+      completed: false,
+      userId: USER_ID,
+      status: status,
+    };
+
+    setTempTodo(fakeTodo);
+
+    postService
+      .postTodo(newTodo)
+      .then(() => {
+        setTodos(prevTodos => [...prevTodos, newTodo]);
+        setShouldFocusInput(true);
+        setTitle('');
+      })
+      .catch(() => {
+        handleError('Unable to add a todo', setErrorMessage);
+      })
+      .finally(() => {
+        setInputDisabled(false);
+        setTempTodo(null);
+      });
+  };
+
+  const updateTodo = (patchTodo: Todo) => {
+    postService.updateTodo(patchTodo).then(task => {
+      setTodos(currentTodos => {
+        const newTodos = [...currentTodos];
+        const index = newTodos.findIndex(({ id }) => id === task.id);
+
+        newTodos[index] = task;
+
+        return newTodos;
+      });
+    });
+  };
+
+  const deleteTodo = (id: number) => {
+    setLoading(true);
+    postService
+      .deleteTodo(`/todos/${id}`)
+      .then(() => {
+        setTodos(prevTodos => prevTodos.filter(todoItem => todoItem.id !== id));
+      })
+      .catch(() => {
+        handleError('Unable to delete a todo', setErrorMessage);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setLoading(false);
+      });
+  };
+
+  const handleClearCompleted = () => {
+    const completedTodoIds = todos
+      .filter(duty => duty.completed)
+      .map(duty => duty.id);
+
+    completedTodoIds.forEach(id => {
+      postService
+        .deleteTodo(`/todos/${id}`)
```

Comment:
```suggestion
        .deleteTodo(id)
```
`handleClearCompleted` shouldn't know anything about URL, pass only id and manage URL in the service function

Reply:
not fixed